rdt/hyper_transformer.py

```diff
@@ -15,11 +15,132 @@
 class Config(dict):
     """Config dict for ``HyperTransformer`` with a better representation."""
 
+    _provided_field_transformers = None
+    _provided_field_sdtypes = None
+
+    @staticmethod
+    def _validate_config(config):
+        sdtypes = config.get('sdtypes', {})
+        transformers = config.get('transformers', {})
+        for column, transformer in transformers.items():
+            input_sdtype = transformer.get_input_sdtype()
+            sdtype = sdtypes.get(column)
+            if input_sdtype != sdtype:
+                warnings.warn(
+                    f'You are assigning a {input_sdtype} transformer to a {sdtype} '
+                    f"column ('{column}'). If the transformer doesn't match the "
+                    'sdtype, it may lead to errors.'
+                )
+
+    @staticmethod
+    def _get_supported_sdtypes():
+        return get_transformers_by_type().keys()
+
+    def set_config(self, config):
+        """Set the ``HyperTransformer`` configuration.
+
+        This method will only update the sdtypes/transformers passed. Other previously
+        learned sdtypes/transformers will not be affected.
+
+        Args:
+            config (dict):
+                A dictionary containing the following two dictionaries:
+                - sdtypes: A dictionary mapping column names to their ``sdtypes``.
+                - transformers: A dictionary mapping column names to their transformer instances.
+        """
+        self._validate_config(config)
+        self._provided_field_sdtypes = config['sdtypes']
+        self['field_sdtypes'].update(config['sdtypes'])
+        self._provided_field_transformers = config['transformers']
+        self['field_transformers'].update(config['transformers'])
+
+    def __init__(self, config=None):
+        super().__init__()
```

Comment:
The class probably no longer needs to inherit from `dict`. It could take a dict in the `__init__` and set things up accordingly. This would discourage people from using the class like a dict and encourage them to use the setter methods we create instead